apps/condo/domains/resident/utils/serverSchema/findOrganizationsByAddress.js

```diff
@@ -11,112 +8,265 @@ const {
+
+    const groupedMeters = await getOrganizationGroupedMeters(organizations, addressKey, { unitName, unitType })
+
+    return organizations.map(organization => {
+        let receipts = groupedReceipts[organization.id]
+        let meters = groupedMeters[organization.id]
+
+        if (receipts?.length) {
+            const cleanedReceipts = {}
+
+            for (const receipt of receipts) {
+                if (!cleanedReceipts[receipt.category]){
+                    cleanedReceipts[receipt.category] = receipt
+                } else {
+                    cleanedReceipts[receipt.category] = { category: receipt.category }
                 }
             }
+
+            receipts = Object.values(cleanedReceipts)
+        }
+
+        if (blackListIndex[organization.id]){
+            if (receipts?.length){
+                receipts = receipts.map(({ category }) => ({ category }))
+            }
+            if (meters?.length){
+                meters = meters.map(({ resource }) => ({ resource }))
+            }
         }
-        const accountNumber = accountQuery.number
-        if (!receipts.length && checkAccountNumberUrl && accountNumber) {
+
+        return {
+            id: organization.id,
+            name: organization.name,
+            tin: organization.tin,
+            type: organization.type,
+            receipts: receipts,
+            meters: meters,
+        }
+    })
+}
+
+async function findOrganizationsByAddressKeyTinAccountNumber (organizations, billingContextIndex, { addressKey, tin, accountNumber }) {
+    const billingIntegrationsWithRemoteInteraction = await find('BillingIntegration', {
+        id_in: Object.values(billingContextIndex).map(({ integration }) => integration),
+        checkAccountNumberUrl_not: null,
+    })
+
+    const remoteInteractions = Object.fromEntries(
+        billingIntegrationsWithRemoteInteraction.map(({ id, checkAccountNumberUrl }) => ([id, checkAccountNumberUrl]))
+    )
+
+    const receipts = await Promise.all(organizations.map(async organization => {
+        const context = billingContextIndex[organization.id]
+        if (!context) return null
+
+        const billingAccounts = await find('BillingAccount', {
+            context: { id: context.id },
+            deletedAt: null,
+            number: accountNumber,
+        })
+
+        let receipts = []
+        if (billingAccounts.length && context.lastReport?.period) {
+            receipts = await find('BillingReceipt', {
+                context: { id: context.id },
+                deletedAt: null,
+                period: context.lastReport.period,
+                account: { id_in: billingAccounts.map(({ id }) => id) },
+            })
+        }
+
+        const checkAccountNumberUrl = remoteInteractions[context.integration]
+
+        if (!receipts.length && checkAccountNumberUrl) {
             const { status, services } = await getAccountsWithOnlineInteractionUrl(checkAccountNumberUrl, tin, accountNumber)
```

Comment:
getAccountsWithOnlineInteractionUrl - will return the same output type
FindOrganizationByAddressReceiptType, FindOrganizationByAddressMeterType